src/main/java/edu/harvard/iq/dataverse/util/FileUtil.java

```diff
@@ -1484,7 +1487,18 @@ public static void validateDataFileChecksum(DataFile dataFile) throws IOExceptio
         try {
             recalculatedChecksum = FileUtil.calculateChecksum(in, checksumType);
         } catch (RuntimeException rte) {
+            logger.log(Level.SEVERE, "failed to calculated checksum, one retry", rte);
             recalculatedChecksum = null;
+            IOUtils.closeQuietly(in);
+            storage = dataFile.getStorageIO();
+            try {
+                storage.open(DataAccessOption.READ_ACCESS);
```

Comment:
If storage.open is part of the get stream call, could you use try with resources here? (I think your code works but having the outer try close the stream you open in the inner try's catch seems fragile. Plus we're slowly shifting to use try with resources in general.)

Reply:
I refactored the try-catch blocks to the try-with-resources. I think the code is not only safer but also more readable right now. Thanks!